Source/Core/Common/Debug/MemoryPatches.cpp

```diff
@@ -90,6 +98,87 @@ void MemoryPatches::RemovePatch(std::size_t index)
   m_patches.erase(m_patches.begin() + index);
 }
 
+void MemoryPatches::LoadFromStrings(const std::vector<std::string>& patches)
+{
+  for (const std::string& patch : patches)
+  {
+    std::stringstream ss;
+    ss.imbue(std::locale::classic());
+
+    // Get the patch state (on, off)
+    std::string state;
+    ss << patch;
+    ss >> state;
+    const bool is_enabled = state == "on";
+    if (!ss)
+      continue;
+
+    // Get the patch address
+    u32 address;
+    ss >> std::hex >> address;
+    ss >> std::ws;
+    if (!ss)
+      continue;
+
+    // Get the patch value
+    std::string hexstring;
+    ss >> hexstring;
+    if (!ss)
+      continue;
+
+    // Check the end of line
+    std::string is_not_eol;
+    if (ss >> std::ws >> is_not_eol)
+      continue;
+
+    const bool is_hex_valid =
+        hexstring.find_first_not_of("0123456789abcdefABCDEF") == hexstring.npos &&
+        (hexstring.size() % 2) == 0;
+    if (!is_hex_valid)
+      continue;
+
+    // Convert the patch value to bytes
+    std::vector<u8> value;
+    const std::size_t len = hexstring.length();
+    for (std::size_t i = 0; i < len; i += 2)
+    {
+      std::size_t size;
+      u32 hex = std::stoi(hexstring.substr(i, 2), &size, 16);
+      if (size != 2)
+      {
+        value.clear();
+        break;
+      }
+      value.push_back(static_cast<u8>(hex));
+    }
+
+    if (value.empty())
+      continue;
+
+    const std::size_t index = m_patches.size();
+    SetPatch(address, value);
+    if (!is_enabled)
+      DisablePatch(index);
+  }
+}
+
+std::vector<std::string> MemoryPatches::SaveToStrings() const
+{
+  std::vector<std::string> patches;
+  for (const auto& patch : m_patches)
+  {
+    std::ostringstream oss;
```

Comment:
I wonder if it makes sense to make a `fmt::formatter` for `MemoryPatch` then  you just do:  `fmt::to_string(patch)`...

Reply:
Another option is to implement `std::ostream& operator<<` and use `fmt`'s [ostream support](https://fmt.dev/latest/api.html#ostream-api), if that'd make it easier to be consistent with the parsing code.